### PR TITLE
WIP - Add messages

### PR DESCRIPTION
**Pull request by nramadas**
This is just a work in progress. I'm hoping this can go into the product while I wait on UX/design updates and move on to something else in the meantime.

:eyeglasses: @schwers @phil303

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,4 +7,4 @@ bin/*
 build/
 .test/
 .DS_Store
-
+reddit-mobile.log
```
Comment (phil303) [inline .gitignore]: just curious, what is this?

File: src/app/components/DirectMessage/index.jsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  recipient: T.string,
+};
+
+DirectMessage.defaultProps = {
+  recipient: '',
+};
+
+const selector = createSelector(
+  state => state.user,
+  (user) => ({ user })
```
Comment (uzi) [inline src/app/components/DirectMessage/index.jsx]: Are the parenthesis needed on either of these?

File: src/app/components/Messages/Comment.jsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+export default function MessagesComment(props) {
+  const { comment } = props;
+
+  return (
+    <div className='MessagesComment'>
+      <div className='MessagesComment__title'>
+        { comment.linkTitle }
+      </div>
+      <div className='MessagesComment__metaData'>
+        { `${ comment.author } • r/${ comment.subreddit } • ${ short(comment.createdUTC) }` }
```
Comment (uzi) [inline src/app/components/Messages/Comment.jsx]: Is there a reason why we'd want to do something like the hex or html version of those dots instead of the unicode character?

File: src/app/components/Messages/Header.jsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  return (
+    <div className='MessageHeader'>
+      <div className='MessageHeader__title'>Inbox</div>
+      {/*<Anchor
+        className='MessageHeader__compose icon icon-message'
+        href='/message/compose'
+      />*/}
+    </div>
+  );
+}
```
Comment (uzi) [inline src/app/components/Messages/Header.jsx]: Commented stuff in this file.  Remove or maybe we want to add comments about it?

File: src/app/components/Messages/Message.jsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  const { message } = props;
+
+  return (
+    <div className='MessagesMessage'>
+      <div className='MessagesMessage__header'>
+        <div className='MessagesMessage__title'>
+          { message.subject }
+        </div>
+        <div className='MessagesMessage__subtitle'>
+          { `${ message.author } • ${ short(message.createdUTC) }` }
```
Comment (uzi) [inline src/app/components/Messages/Message.jsx]: Same unicode character question

File: src/app/components/Messages/Message.less [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+  &__body {
+    font-size: 11px;
+    line-height: 1.2;
+
+    .themeify({
+      color: @theme-body-text-color;
+    });
+
+    a {
```
Comment (schwers) [inline src/app/components/Messages/Message.less]: Can the styles in [text.less](https://github.com/reddit/reddit-mobile/blob/2X/src/app/less/components/text.less#L9) handle this?

File: src/app/components/Messages/index.jsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+  mailType: T.string.isRequired,
+};
+
+const renderItem = (type, id, messages, comments, posts) => {
+  switch (type) {
+    case 'comment': return <MessagesComment comment={ comments[id] }/>;
+    case 'message': return <MessagesMessage message={ messages[id] }/>;
+    case 'post': return <MessagesPost post={ posts[id] }/>;
+    default: return null;
+  }
```
Comment (uzi) [inline src/app/components/Messages/index.jsx]: I was about to brain fart and ask if you needed breaks ... you're returning in all cases, so you don't.  Ummm... I usually like the formatting of

```
    case 'foo':
        return something();
    case 'bar':
        return somethingElse();
    ...
```

But that's a personal stylistic choice.  I just like whitespace.

File: src/app/components/DirectMessage/index.jsx (hunk 2 above)
Comment (nramadas) [inline src/app/components/DirectMessage/index.jsx]: they are needed on the second. i can remove them on the first.

File: src/app/components/DirectMessage/index.jsx [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+export function DirectMessage(props) {
+  return (
+    <div className='DirectMessage'>
+      <div className='DirectMessage__header'>
+        <DirectMessageHeader/>
+      </div>
+      <div className='DirectMessage__body'>
+        <DirectMessageComposition
+          username='nr4madas'
```
Comment (schwers) [inline src/app/components/DirectMessage/index.jsx]: Should we add a todo for this component and un-hardcode your username?

File: src/app/components/Messages/Message.less (hunk 6 above)
Comment (nramadas) [inline src/app/components/Messages/Message.less]: i didn't know that was a thing! i'll see if i can use text.less

File: .gitignore (hunk 1 above)
Comment (nramadas) [inline .gitignore]: not sure. it started appearing in my commits (and they seemed totally useless) so i'm gitignoring them for now. I'll spend some time later trying to figure out what it does.

File: src/app/components/DirectMessage/index.jsx (hunk 8 above)
Comment (nramadas) [inline src/app/components/DirectMessage/index.jsx]: yes. we definitely should.

Comment (phil303) [inline src/app/components/DirectMessage/index.jsx]: replace it with your password plz

File: .gitignore (hunk 1 above)
Comment (ajacksified) [inline .gitignore]: https://github.com/reddit/reddit-mobile/blob/8767a9747f4a5e03091b0f9729db1a298131a214/src/server/console.es6.js#L26

logs requests when you use the ascii console.

File: src/app/components/Messages/Comment.less [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+  &__body {
+    font-size: 11px;
+    line-height: 1.2;
+
+    .themeify({
+      color: @theme-body-text-color;
+    });
+
+    a {
```
Comment (schwers) [inline src/app/components/Messages/Comment.less]: Same text.less question

File: src/app/components/Messages/Header.jsx (hunk 4 above)
Comment (nramadas) [inline src/app/components/Messages/Header.jsx]: added a comment

File: src/app/reducers/mail.js [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+    case mailActions.PENDING: {
+      const { mailType } = action;
+      return merge(state, {
+        [mailType]: {
+          pending: true,
+          error: null,
+        },
+      });
+    }
+    case mailActions.SUCCESS: {
```
Comment (schwers) [inline src/app/reducers/mail.js]: I think we should stick with `FETCHING` and `RECEIVED` convention the other files are using just for consistency. Happy to switch up the names but seems like that'd be good in a separate patch.

**Comment by schwers** [comment]
💅

File: .gitignore (hunk 1 above)
Comment (phil303) [inline .gitignore]: cool thanks for the info

File: src/app/router/handlers/Messages.js [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+import { apiOptionsFromState } from 'lib/apiOptionsFromState';
+import * as mailActions from 'app/actions/mail';
+
+export default class Messages extends BaseHandler {
+  async [METHODS.GET](dispatch, getState/*, utils*/) {
+    const { mailType } = this.urlParams;
+    const apiOptions = apiOptionsFromState(getState());
+
+    fetchUserBasedData(dispatch);
+    dispatch(mailActions.setInboxPending(mailType));
```
Comment (phil303) [inline src/app/router/handlers/Messages.js]: I have to look at the platform source again but would it make sense to move this into an action creator in `actions/mail.js`? The goal here being to maintain more consistency with other action files in our code base (at least the mweb one).

File: src/app/components/SettingsOverlayMenu/index.jsx [hunk 12]
```diff
@@ -67,7 +67,7 @@ export const renderLoggedInUserRows = (user) => {
     <LinkRow
       key='inbox'
       text={ ['Inbox', badge] }
-      href='/message/inbox'
+      href='/message/messages'
       icon={ `icon-message icon-large ${inboxCount ? 'orangered' : 'blue'}` }
```
Comment (nramadas) [inline src/app/components/SettingsOverlayMenu/index.jsx]: sure

**Comment by nramadas** [comment]
:haircut:

File: src/app/components/Messages/Post.jsx [hunk 13]
```diff
@@ -0,0 +1,9 @@
+import './Post.less';
+import React from 'react';
+
+export default function MessagesPost() {
+  return (
+    <div className='MessagesPost'>
```
Comment (schwers) [inline src/app/components/Messages/Post.jsx]: Could you add a note or todo here explaining what needs to be done in the Post case?

Comment (nramadas) [inline src/app/components/Messages/Post.jsx]: sure

File: src/app/reducers/posts.js [hunk 14]
```diff
@@ -26,7 +27,8 @@ export default function(state=DEFAULT, action={}) {
     case postsListActions.RECEIVED_POSTS_LIST:
     case hiddenActions.RECEIVED_HIDDEN:
     case savedActions.RECEIVED_SAVED:
-    case searchActions.RECEIVED_SEARCH_REQUEST: {
+    case searchActions.RECEIVED_SEARCH_REQUEST:
+    case mailActions.SUCCESS: {
```
Comment (schwers) [inline src/app/reducers/posts.js]: It looks like not all of the `success` constants were converted to `received`. I get an empty inbox at the moment but it looks like its just because the reducers aren't pulling in the data

**Comment by schwers** [comment]
Just two small things, looks good! 💅

File: lib/snooboots [hunk 15]
```diff
@@ -0,0 +1 @@
+Subproject commit 50555e5c9894a91bc581338b584f9c8cfd982bc7
```
Comment (schwers) [inline lib/snooboots]: Whoops, I should have caught this earlier, but it looks like you're going to have to remove `lib/snooboots` and `lib/art-gallery`.

**Comment by nramadas** [comment]
:haircut:

**Comment by schwers** [comment]
:fish:
